File: ChAvTicks.Application/Responses/Airport/Common/AirportSummaryResponse.cs
using ChAvTicks.Application.Responses.Base;
using System.ComponentModel.DataAnnotations;

namespace ChAvTicks.Application.Responses.Airport.Common
{
    [Serializable]
    public sealed record AirportSummaryResponse(
            string? Icao,
            string? Iata,
            string? LocalCode,
            string? ShortName,
            [Required] string? FullName,
            string? MunicipalityName,
            AirportLocationResponse? Location,
            string? CountryCode,
            string? CountryName)
        : AirportResponseBase(
            Icao,
            Iata,
            LocalCode,
            ShortName,
            MunicipalityName,
            Location);
}

File: chavticks.ui/src/app/responses/airport/AirportSummaryResponse.ts
import {AirportLocationResponse} from "./AirportLocationResponse";

export class AirportSummaryResponse {
  icao: string | null;
  iata: string | null;
  localCode: string | null;
  name: string;
  shortName: string | null;
  municipalityName: string | null;
  location: AirportLocationResponse;
  countryCode: string | null;
  country: string | null;
}
